main.go

```diff
@@ -225,6 +221,61 @@ func Start(id uint64, acquirer acquire.Acquirer, logger log.Logger, requestURL s
 
 }
 
+func createWrp(message Message, fixedCurrentTime func() time.Time, logger log.Logger) wrp.Message {
+	var current time.Time
+	if message.FixedCurrentTime {
+		current = fixedCurrentTime()
+	} else {
+		current = time.Now()
+	}
+
+	wrpMsg := message.Wrp
+	wrpMsg.Type = 4
+
+	if len(wrpMsg.Destination) == 0 {
```

Comment:
I think caduceator might be a bit more performant if some of these validation checks were done once to the message at startup rather than with each message creation, but I don't feel that strongly about this.